Add al-ko to latest

zechnerhubert opened the pull request.

zechnerhubert committed:
add(al-ko): include adapter in latest repository (type garden)

mcm1957 commented:
reminder 5.11.2025

mcm1957 commented:
from comment above:

To verify the object structure of this adapter during REVIEW **please export the object structure of a working installation and attach the file to this PR**. You find a guide how to export the object struture here: https://github.com/ioBroker/ioBroker.repochecker/blob/master/OBJECTDUMP.md

reminder 11.11.2025

zechnerhubert commented:
[al-ko.json](https://github.com/user-attachments/files/23345367/al-ko.json)

mcm1957 commented:
@zechnerhubert 

First of all - THANK YOU for the time and effort you spend to maintain this adapter.

I would like to give some feedback based on my personal oppinion. @Apollon77 might have additional suggestions or even a different oppinion to one or the other statement. Please feel free to contact him if you cannot follow my suggestions or want to discuss some special aspects.

- [ ] errors reported by adapter checker must be fixed

  Errors and warnings as far as possible reported by repository checker must be fixed. Please add missin size attributes to jsonConfig components and verify responsive design. For details see https://github.com/iobroker-community-adapters/responsive-design-initiative

- [x] Readme.md must be written in pure english

  As ioBroker is an international software supporting several languages the main Readme.md file should be written in english. Its ok to add additional Readme-<lang>.mds of course.
  
- [x] Readme.md contains multiple languages
  
  Please remove the german version / all non english versions from Readme.md because Readme.md gets automatically translated and this will not work with multiple languages within Readme.md. You can add locatlized Readme pages in (docs/de and docs/en) like https://github.com/iobroker-community-adapters/ioBroker.pvforecast/tree/main/docs and add it to io-package (see https://github.com/iobroker-community-adapters/ioBroker.pvforecast/blob/main/io-package.json#L119-L122)
 
- [x] Is this adapter officially supported by AL-KO ?

  If this adapter supported by al-ko? If this is not the case I would suggest to remove the contributors entry at package.json. This suggests that alco has worked on this adapter. This could lead to legal problems. see https://github.com/zechnerhubert/ioBroker.al-ko/blob/97f80328624bb71e7103f484bbb0f0bbbc717b89/package.json#L11 

- [ ] unneeded dependencies

  As you use @iobroker/eslint-config no local includes of eslint / prettier packages are required. They should be removed
    "eslint": "^9.39.0",
    "prettier": "^3.3.3",
    "@eslint/js": "^9.14.0",
    "typescript-eslint": "^8.8.1

  see https://github.com/ioBroker/eslint-config/blob/main/MIGRATION.md

- [x] log messages should / must be in english

  As ioBroker is an international software, please use only english wording for all logging messages - unless you translate logging depending on system language. Logging using the system langugae is ok of course - buts its your decision if you want toe invest the effort. German only logs are not desired. Thios refers to all text defined by you - error messages reurned from aservice (external text) can be kepogt as is of course.

- [x] log messages must not have beautifiers

  Please remove all beutiofiers from log messages. Logs should aonly contain ascii characters  unless they are a unchanged output of text received from an external source.

- [x] plese minimize logging at info level and output only messages which are really important and occure unfrequently.

  i.e its not useful to log multiple infos for ony state chagne: https://github.com/zechnerhubert/ioBroker.al-ko/blob/97f80328624bb71e7103f484bbb0f0bbbc717b89/main.js#L446 . Using level bebug would be appropiate there

- [x] hardcoded instance value ?

  Looks like the adapter hardcodes the instance value to 0. This is invalid. If there is no special reason to limit the adapter to one instance it shoudl support any number of instances. But even if the adapter is correctly configured as singleton there is no force that the instance number is 0. In addition normally it is not required to handle instance number explizitly as setObjectNotExists, setState etc. always refer to teh own instance per default. Only when using setStateForeign etc. a full id is required.

  see https://github.com/zechnerhubert/ioBroker.al-ko/blob/97f80328624bb71e7103f484bbb0f0bbbc717b89/main.js#L155

- [x] invalid characters should be filtered from object ids
  
  Some characters are not allowed to be used as part of an object id. If an object id is not hardcoded or even depending on user input you should ensure that such an object id does not contain an invalid character.  Iobroker provides the constant ```adapter.FORBIDDEN_CHARS```, i.e. by using the following snippet:
  ```
  function name2id(pName) {
    return (pName || '').replace(adapter.FORBIDDEN_CHARS, '_');
  }
  ```

  You are retrieving the device id form an external source. (https://github.com/zechnerhubert/ioBroker.al-ko/blob/97f80328624bb71e7103f484bbb0f0bbbc717b89/main.js#L138) so you must process the received data and ensure that this string does noz conatin invalid characters. At leasteFORBIDDEN_CHARS must be removed (or replaced). In addition spaces and dots should be replaced as spaces are valid for historic reasons but are known to cause problems with some vis implementations. And dots need a special handling as the are used to seperate object leveös (aka 'folders'). Bet praxis is to limit state ids to [A-Za-z0-9_-]

- [ ] onStateChange handler ignores ack flag 

  The adapter must honor the ack flag when processing onStateChange events. Whenever the onStateChange handler is called for an own state (a state owned by the processing instance of the adapter) it must ignore this event if the ack flags is set. Only stateChanges with ack==false are allowed to be processed.

  When processing foreign states which are output of another adapter  processing must only occure if the ack flag is true.

- [ ] creation of intermediate objects missing

  Intermediate objects (device, channels, folders) are not created. I.e no object exists for 'details', 'info', ...

  The main device object (identified by some OID) and teh folder (?) state are not created:

<img width="1259" height="200" alt="Image" src="https://github.com/user-attachments/assets/e8f96331-2c61-46a1-ab54-a43b14cd64fc" />

- [ ] prupose of filder state

  Whats the purpose of folder state? It's alone at its level. Is this object really required / useful?

- [ ] reevaluate state roles

  Only the values specified here (https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/stateroles.md) may be used as state roles. Do not invent new names as this might disturb the functionalyity of other adapters or vis.

  In addition the roles MUST match the read/write functionality. As an example a role value.* requires that the state is a readable state. Input states (write only) should have roles like level.*. Please read the description carefully. States with role 'button' should (must) have attribute 'common.read=false' set. A button ( "Taster" in german only triggers when you press but else has no state), so it should also be not readable. This is also like HomeMatic does it. A switch has clear states true or false and so can be read.

  Please avoid using general roles (state, value) whenever a dedicated role exists.

- [ ] some labels seem to have no translation into all languages
  
  It seems that some labels are not translated into all supported languages
  Please consider adding missing translations
  
- [ ] Is there any reason to not use i18n translation environment?
  
  I suggest to evaluate to use i18n translation environment and the iobroker/adapter-dev environment (https://github.com/ioBroker/adapter-dev#iobroker-adapter-dev). You could easyly add new strings to the english json and run ```npm run translate``` afterward. When using i18n environment activating iobrokers weblate translation system (see https://github.com/ioBrokerTranslator/doc) is trivial.  

  If you really prefer to use local translations within jsonConfig (NOT recommended) you must provide translations for all supported languges for all  components.

- [x]  jsonConfig size attrribute is invalid
  soie: [xs:nn, ...} is an invlid syntax for jsonconfig. 
  https://github.com/zechnerhubert/ioBroker.al-ko/blob/97f80328624bb71e7103f484bbb0f0bbbc717b89/admin/jsonConfig.json#L36

  Please see jsonConfig dokumantation https://github.com/ioBroker/ioBroker.admin/blob/master/packages/jsonConfig/README.md

- [x] Add timeout to axios calls

  All axios calls should have a timeout set. The default value for axios timeout is 0 which results in no timeout. So an unresponsive remote system could lead to an hanging adapter. Either set an appropiate timeout per axios call or set a default timeout globally for the axios instance.
  
- [x] consider using adapter.setTimeout / this.setTimeout instead of (standard) setTimeout
  
  The adapter package provides wrapper routines for native setTimeout, setInterval, clearTimeout and clearInterval. Using those routines ensures that timers are cancelled on on load. Additional checks on thomse limits might be performed, too. So consider replacing native setTimeout/clearTimeout by adapter.setTimeout/adapter.clearTimeout or this.setTimeout/this.clearTimeout. The same refers to setInterval/clearInterval.

- [x] check and adapt testing
  
  Consider activating integrated deploy to npm by uncommenting the corresponding section at test-and-release workflow.
  (Suggestion only, not required)
  
  Plese ensure that the test-and.release workflow is run after every commit and for every release tag settings

Thanks for reading and evaluating this suggestions.
McM1957

**Please add a comment when you have reviewed and fixed the suggestionsor at least commented the suggestions and you think the adapter is ready for a re-review!**

**Please add a new object dump after reviewing state roles and ob ject creation**

reminder 25.11.2025

mcm1957 commented:
@zechnerhubert 

Hi,
Thanks for your work on an ioBroker Adapter.

**Did you notice my feedback from 11.11.2025?**

Please let me know if you already analyzed / fixed the issues reported. If not pleas let me know an estimated timeframe when a update can be expected.

reminder 17.12.2025

zechnerhubert commented:
Hi Martin,

thanks for the reminder and for your feedback from 11.11.2025.

I’ve seen your notes and I’m currently working through the remaining points. Progress is a bit slower right now because the mower (hardware) won’t be available for real-world testing again until April/May 2026 (start of the mowing season). Until then, I’ll continue improving the adapter to be fully ioBroker-conform (structure, roles, linting/adapter-check compliance, etc.) and prepare the changes so they can be tested and released as soon as the device is available again.

So, a tested update is realistically expected around April/May 2026. If I can safely release an interim update earlier (non-breaking, mainly conformity/cleanup), I’ll do that.

Thanks again for your patience and support.

Best regards
Hubert

 

 

Von: Martin M. ***@***.***> 
Gesendet: Mittwoch, 3. Dezember 2025 15:26
An: ioBroker/ioBroker.repositories ***@***.***>
Cc: hubertiob ***@***.***>; Mention ***@***.***>
Betreff: Re: [ioBroker/ioBroker.repositories] Add al-ko to latest (PR #5257)

 

  <https://avatars.githubusercontent.com/u/28575778?s=20&v=4> mcm1957 left a comment (ioBroker/ioBroker.repositories#5257) <https://github.com/ioBroker/ioBroker.repositories/pull/5257#issuecomment-3607121105> 

@zechnerhubert <https://github.com/zechnerhubert> 

Hi,
Thanks for your work on an ioBroker Adapter.

Did you notice my feedback from 11.11.2025?

Please let me know if you already analyzed / fixed the issues reported. If not pleas let me know an estimated timeframe when a update can be expected.

reminder 17.12.2025

—
Reply to this email directly, view it on GitHub <https://github.com/ioBroker/ioBroker.repositories/pull/5257#issuecomment-3607121105> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AG3BWOXQXVZ7TUBOT3H3UQL373XHNAVCNFSM6AAAAACKSTKFGCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTMMBXGEZDCMJQGU> .
You are receiving this because you were mentioned.  <https://github.com/notifications/beacon/AG3BWOUYAVPEQJL3B3WMM6D373XHNA5CNFSM6AAAAACKSTKFGCWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTWXABGNC.gif> Message ID: ***@***.*** ***@***.***> >

mcm1957 commented:
reminder 17.3.2026

zechnerhubert committed:
Merge branch 'master' into add-al-ko-latest

mcm1957 commented:
@zechnerhubert 

Sorry for the long delay. And thnaks a lot for fixing many issues in the meantime.

Please review thos issues still open:

- [x] errors reported by adapter checker must be fixed

  Errors and warnings as far as possible reported by repository checker must be fixed. Please add missin size attributes to jsonConfig components and verify responsive design. For details see https://github.com/iobroker-community-adapters/responsive-design-initiative

  Please fix:
  - 👀 [W5048] ".eslintrc.json" is most likely obsolete when using "@iobroker/eslint-config". Please remove it.
     Removing the file should fix this

  - 👀 [W5512] admin/jsonConfig.json schema validation error: Unknown property "label" in root
    If you do not have an tabs to display the label, there is no label attribute. Please remove.

  - 👀 [W6017] CHANGELOG.md detected. The changelog must be located within README.md
    Please move changelog into README.md for the most actual releases. Older Changelog can (and should) be placed in a file named CHANGELOG_OLD.md. Nameing the file this was it will be detected and used by @alcalzone/releasescritp tool.

- [x] unneeded dependencies

  As you use @iobroker/eslint-config no local includes of eslint / prettier packages are required. They should be removed
    "prettier": "^3.3.3",
  also
     "proxyquire": "^2.1.3"
  is most likely not used ans can be removed.

- [x] onStateChange handler ignores ack flag 

  The adapter must honor the ack flag when processing onStateChange events. Whenever the onStateChange handler is called for an own state (a state owned by the processing instance of the adapter) it must ignore this event if the ack flags is set. Only stateChanges with ack==false are allowed to be processed.

  When processing foreign states which are output of another adapter  processing must only occure if the ack flag is true.

  If you have questions, feel free to ask.

- [ ] creation of intermediate objects missing

  **Please attach a new object dump so that a check if this issue is fixed already can be performed.**

  Intermediate objects (device, channels, folders) are not created. I.e no object exists for 'details', 'info', ...

  The main device object (identified by some OID) and teh folder (?) state are not created:

  <img width="1259" height="200" alt="Image" src="https://github.com/user-attachments/assets/e8f96331-2c61-46a1-ab54-a43b14cd64fc" />


- [ ] purpose of folder state

  **Please attach a new object dump so that a check if this issue is fixed already can be performed.**

  Whats the purpose of folder state? It's alone at its level. Is this object really required / useful?

 
- [ ] reevaluate state roles

  **Please attach a new object dump so that a check if this issue is fixed already can be performed.**

  Only the values specified here (https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/stateroles.md) may be used as state roles. Do not invent new names as this might disturb the functionalyity of other adapters or vis.

  In addition the roles MUST match the read/write functionality. As an example a role value.* requires that the state is a readable state. Input states (write only) should have roles like level.*. Please read the description carefully. States with role 'button' should (must) have attribute 'common.read=false' set. A button ( "Taster" in german only triggers when you press but else has no state), so it should also be not readable. This is also like HomeMatic does it. A switch has clear states true or false and so can be read.

  Please avoid using general roles (state, value) whenever a dedicated role exists.

- [ ] some labels seem to have no translation into all languages
  
  It seems that some labels are not translated into all supported languages
  Please consider adding missing translations

  You mjust either hardcode ztranslations for all languges withon jsonConfig.json or use i18n environment. To generate hardcodes translation blocks you can use https://translator-ui.iobroker.in/. But the recommended way is to use i18n environment. You cann fill in the translations inti admin/i18n/en.json and let adapterdevs npom run translate do the translations for you. See next topic.

- [ ] Is there any reason to not use i18n translation environment?
  
  I suggest to evaluate to use i18n translation environment and the iobroker/adapter-dev environment (https://github.com/ioBroker/adapter-dev#iobroker-adapter-dev). You could easyly add new strings to the english json and run ```npm run translate``` afterward. When using i18n environment activating iobrokers weblate translation system (see https://github.com/ioBrokerTranslator/doc) is trivial.  

  If you really prefer to use local translations within jsonConfig (NOT recommended) you must provide translations for all supported languges for all  components.

  If you need any help in solving the issues please ask here or at our development channels at telegram / discor. Invite links are available at www.iobroker.dev.

Thanks for reading and evaluating this suggestions.
McM1957

**Please add a comment when you have reviewed and fixed the suggestionsor at least commented the suggestions and you think the adapter is ready for a re-review!**

**Please add a new object dump after reviewing state roles and ob ject creation**

reminder 25.4.2026

zechnerhubert commented:
Hi,

I have reviewed all comments and implemented the requested changes:

* Fixed object creation and ensured proper device/channel/state structure
* Added intermediate objects where required
* Implemented dynamic role and type assignment for all states
* Recreated all objects to apply correct metadata
* Migrated admin configuration to the i18n environment
* Adjusted changelog handling according to the current ioBroker requirements (latest entries in README, older entries separated)

A fresh object dump is attached for verification.

Regarding warning W5022 (i18n):
The adapter has been fully migrated to the i18n environment. All translation files exist for all supported languages and contain identical keys.
Despite this, the warning still appears. From my perspective, this might be a false positive or related to a previous state.

Please let me know if I am missing something here.

The adapter should now be ready for re-review.

Thanks!
[al.ko_objects.txt](https://github.com/user-attachments/files/26655150/al.ko_objects.txt)

mcm1957 commented:
@zechnerhubert 

Thanks for fixing the issues and your patience.

- Please attach a new object dump as described above to check corrections to iobejt structure. Don't know how you created the text file - but this is not really useable for me. Please provide json as described and done at the first pass. (Sorry for requesting this work to be done by you once more).

- The i18n issue is caused by missing i18n attroibute at jsonCOnfig.json. See https://github.com/ioBroker/ioBroker.admin/blob/master/packages/jsonConfig/README.md#i18n---internationalization. Please add 

    "i18n": true

  at top level. Code seems to work without the attribute too but according to documentation i18n must be specified. 

Thanks for your cooperation. As soon as I see the json export I'll recheck you adapterwith top priority.

reminder 28.4.2026

zechnerhubert commented:
Hi,

thanks for your feedback and your patience as well.

I have now addressed the remaining points:

.) Added the missing "i18n": true attribute to jsonConfig.json according to the documentation, which resolved the i18n warning.
.) Attached a new object dump in JSON format as requested for proper verification.

Please let me know if anything is still missing.

Also, a quick question regarding the release process:
Would you recommend creating a new npm version (e.g. 0.3.6) after approval, or should I already prepare/publish it before the final review?

Additionally, I would like to mention that parts of this adapter were developed with AI assistance.
Is it recommended or required to add a note about this in the README, or is this not necessary for ioBroker adapters?

Thanks again for your support!

Best regards

Hubert

[al-ko.0.json](https://github.com/user-attachments/files/26940566/al-ko.0.json)

mcm1957 commented:
@zechnerhubert 

Sorry for the long delay. And thnaks a lot for fixing many issues in the meantime.

Please review those issues still open:

- [x] creation of intermediate objects missing

  Intermediate objects (device, channels, folders) are not created. I.e no object exists for 'details', 'info', ...

  The main device object (identified by some OID) and the folder (?) state are not created:

  <img width="1259" height="200" alt="Image" src="https://github.com/user-attachments/assets/e8f96331-2c61-46a1-ab54-a43b14cd64fc" />

  The (folder ?) state is still not created. There is no object defined. If you think it is created somewhere, please indicate where...

  <img width="1219" height="343" alt="image" src="https://github.com/user-attachments/assets/f41babcb-32b4-4ad2-9925-5c7ffff46f0a" />
 
- [x] purpose of folder state

  Whats the purpose of folder state? It's alone at its level. Is this object really required / useful?

  I want to repeat the question. Do you need this folder for any purpose. Looks like it could be omitted. But if you want to keep it an object must be created - most likely of type folder.
 
- [ ] consider adding units

   This is not blocking buit a suggestion. 
   Consider adding units to improve user expirence. i.e Voltage should have V as unit.
  <img width="1421" height="123" alt="image" src="https://github.com/user-attachments/assets/eb168682-7a9b-4ca2-99c6-6a38ee378eac" />

- [ ] reevaluate state roles

  **Please attach a new object dump so that a check if this issue is fixed already can be performed.**

  Only the values specified here (https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/stateroles.md) may be used as state roles. Do not invent new names as this might disturb the functionalyity of other adapters or vis.

  In addition the roles MUST match the read/write functionality. As an example a role value.* requires that the state is a readable state. Input states (write only) should have roles like level.*. Please read the description carefully. States with role 'button' should (must) have attribute 'common.read=false' set. A button ( "Taster" in german only triggers when you press but else has no state), so it should also be not readable. This is also like HomeMatic does it. A switch has clear states true or false and so can be read.

  Please avoid using general roles (state, value) whenever a dedicated role exists.

  Not sure if entrypoints 1 to 9 should be writeable as entrypoint.0 is readonly. But role value requires state to be read only. So eiterh role or r/w attribute needs to be changed.

  <img width="1433" height="204" alt="image" src="https://github.com/user-attachments/assets/5269bfe4-21a1-4847-b7c8-ea2c739c57ed" />

  some more state / roles mismatch. indocator is and value must be read only. If thos states should be writeable you must use coirrect roles, i.e. swicth, level, ...

  <img width="1476" height="244" alt="image" src="https://github.com/user-attachments/assets/7fc54af4-0c4b-4884-ba06-61de70390716" />

- [x] some labels seem to have no translation into all languages
  
  It seems that some labels are not translated into all supported languages
  Please consider adding missing translations

  Do you think this is a frecnh texte !?!?

  <img width="638" height="368" alt="image" src="https://github.com/user-attachments/assets/33ea185f-f462-46ea-b547-4a3069124f8a" />

  Please delete all translation.json files except the english one and recreate translations using npm run translate command. If you let some AI do the translations please check at least the the languge seems to be correct (of course you cannot really check the content) Simply putting english into several files is no solution.
  
  I suggest to evaluate to use i18n translation environment and the iobroker/adapter-dev environment (https://github.com/ioBroker/adapter-dev#iobroker-adapter-dev). You could easyly add new strings to the english json and run ```npm run translate``` afterward. When using i18n environment activating iobrokers weblate translation system (see https://github.com/ioBrokerTranslator/doc) is trivial. 

If you need any help in solving the issues please ask here or at our development channels at telegram / discor. Invite links are available at www.iobroker.dev. You are welcome i.e. at our starters group https://t.me/+gsX-e8k4mLtmZjZk where you can get help fixing issues if you need help.

Thanks for reading and evaluating this suggestions.
McM1957

**Please add a comment when you have reviewed and fixed the suggestionsor at least commented the suggestions and you think the adapter is ready for a re-review!**

**Please add a new object dump after reviewing state roles and ob ject creation**


reminder 5.5.2026

zechnerhubert commented:
Hi,

thanks again for your review.

I have addressed all reported issues:

* Implemented proper creation of intermediate objects (device → channel → state)
* Removed the unnecessary folder state
* Reviewed and corrected all roles:

  * Writable boolean states now use `switch`
  * Read-only boolean states use `indicator`
  * No custom roles are used anymore
* Fixed unit handling:

  * Removed all invalid units from boolean states
  * Removed heuristic unit assignment based on key names
  * Units are now only set where they are clearly defined
* Ensured consistency between roles and read/write flags
* Cleaned up and recreated translations using `npm run translate`

A new JSON object dump is attached reflecting these changes.

Some decisions were made intentionally:

* I did not assign units to fields like `value.time`, because the API does not clearly define whether values are in minutes, hours, etc.
* I kept generic roles like `value.number` where the API does not provide enough context for a more specific role

If you prefer a stricter mapping in some areas, I can adjust it.

Thanks again for your support.

[al-ko.0.json](https://github.com/user-attachments/files/26980769/al-ko.0.json)

mcm1957 commented:
@zechnerhubert 

Please check whether you attached an old object json. Something seems to be wrong.

You write:

>Removed the unnecessary folder state
>
>Reviewed and corrected all roles:
>
>Writable boolean states now use switch

But objects do not match the description:

folder state still exists although it is created correctly (so no blocking issue)
<img width="1461" height="323" alt="image" src="https://github.com/user-attachments/assets/c23fedc2-a2ae-4221-a514-c05fdb86ef62" />

writeable states still use value or indicator:
<img width="1455" height="207" alt="image" src="https://github.com/user-attachments/assets/95df1cbd-2fd8-4551-a413-52d970a0f0d1" />

<img width="1511" height="123" alt="image" src="https://github.com/user-attachments/assets/80ddc203-1c85-4917-a68b-e0d9078e845c" />

If the dump is correct and changes to object creation are still missing, take a break and talk to your AI :-).
Eventually the problem is caused by an older state structure. You should deinstall de adapter / delete all objects and let the recreate to avoid that the code is now correct but you testsystem still uses outdated defnitions as setObjectNotExists does not change existing objects.

Please check correct if required and attach a new object dump

Thanks
mcm1957

zechnerhubert commented:
Hi,
sorry for the confusion in the previous iteration.
Some of the issues were caused by incorrect adjustments during development, including a mistake introduced while working with AI assistance.
I have now reviewed the implementation again and corrected the affected parts.
The whitelist and state handling have been aligned accordingly, and the object structure and roles should now match the expected behavior.
A new JSON object dump is attached reflecting the current state.
Everything should now be consistent with the review requirements, but please let me know if anything still needs adjustment.
Thanks again for your patience and support.
[al-ko.json](https://github.com/user-attachments/files/27020496/al-ko.json)

mcm1957 commented:
@zechnerhubert 

Sorry to repeat once more:

Still some writeabel states have role value.* In addtion value.number is no valid role at all.
Plese see https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/stateroles.md#values-numbers-read-only

<img width="1014" height="245" alt="image" src="https://github.com/user-attachments/assets/6a2ab69e-07e6-4936-b67f-49a628f0f62e" />

<img width="864" height="160" alt="image" src="https://github.com/user-attachments/assets/6f246564-24e9-4162-a4df-17d4fce8b2c6" />

So most of your RO states should ahve role value (except when a matching value.xxx is available)
Writeable numeric states should most likely use level (or level.xxxx if a matching value exists)

Please see documentation for a list of defined roles.
https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/stateroles.md

Please check state roles once again. The rest seems to be ok. Whther you add units or not is uo to your decision and not blocking.

**Please attach a new object dump after correction.**

Thanks
mcm1957

reminder 7.5.2026

zechnerhubert commented:
Hi,

thanks for your feedback.

I have updated the roles according to the specification:

* removed all invalid `value.number` roles
* read-only numeric states now use `value`
* writable numeric states use `level`

A new JSON object dump is attached.

Everything should now comply with the state role definitions. Please let me know if anything still needs adjustment.

Thanks!

[al-ko.json](https://github.com/user-attachments/files/27085352/al-ko.json)

mcm1957 commented:
ok - thanks for correction
lgtm

mcm1957 commented:
This adapter has been released to latest repository and should be visible within 24h maximum.

Please create a thread at https://forum.iobroker.net/category/91/tester titled like "Test Adapter " to collect some user feedback and provide a link to this topic when requesting addition to stable repository later.

Note: If an other testing topic already exists, it' OK to continue using this topic too.

If you do not have write access to TESTER Section of forum, please contact HOMORAN at forum using PN/CHAT
